### DISCUSSION TITLE
Setting random seed before training

### DISCUSSION THREAD

**Discussion by w2kun** [Lightning-AI/pytorch-lightning#9268] [DDP / multi-GPU / multi-node] [answered]
Hi,

Do I need to manually set the random seed to ensure the synchronization of state (e.g. initialized parameters) across processes when using distributed training?

**Comment by awaelchli** [comment, marked as answer]
No, parameters get broadcast on the first time the model performs forward. This is implemented in the PyTorch [DistributedDataParallel](https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html?highlight=distributeddataparallel#torch.nn.parallel.DistributedDataParallel) wrapper and so applies to Lightning as well.
